### DISCUSSION TITLE
why channel instance can cache exchanges info

### DISCUSSION THREAD

**Discussion by Y-Meta** [rabbitmq/rabbitmq-java-client#1233] [Q&A] [answered]
### Describe the bug

First, I created a wrong exchange in the management web, and then created a exchange through the channel instance to report an error code = 406, then I deleted the exchange in the management web and tried to create the exchange through the previous channel instance, but still reported an error 406 showing that the attributes did not match(i know this mean). My question is why the exchange has been deleted? Why does the channel instance not create the switch but still read the old exchange Information?

### Reproduction steps

1.The management web creates a exchange with durable = true name = x
2.Create a channel instance and create a exchange with durable = false through the instance name=x；error code=406
3.Delete x exchange through the management web
4.Create a exchange with durable = false by using the previous channel instance withname=x；still error code=406

### Expected behavior

I intuitively think that every time I go through the channel.exchangeDeclare method, I will interact with mq (even if this channel has already been created), therefore, after I delete the wrong exchange, the next time I create a exchange through the existing channel, it will succeed.

### Additional context

my client version
<dependency>
	<groupId>com.rabbitmq</groupId>
	<artifactId>amqp-client</artifactId>
	<version>5.3.0</version>
</dependency>

RabbitMQ 3.3.5

**Comment by Y-Meta** [comment]
I want to know if there is a cache in the channel, and if there is a cache, where should I find the relevant cache code? Thank you very much!

**Comment by acogoluegnes** [comment, marked as answer]
There's no cache in the channel. The connection recovery mechanism does keep track of what's created though.

The client and broker versions are outdated. Please try again with the latest versions (Java client 5.20.0 and RabbitMQ 3.12.12) to see if the problem persists.

**Reply by Y-Meta** [reply]
OK  I will upgrade it！Thank you

**Reply by Y-Meta** [reply]
I update Java client 5.20.0 Again, the problem seems to be caused by the ShutdownNotifierComponent.isOpen() method,  shutdownCause is not null. Therefore, after the channel is successfully created, will the channel be closed after executing the exchangeDeclare method that reported an error?

**Reply by Y-Meta** [reply]
![error](https://github.com/rabbitmq/rabbitmq-java-client/assets/61530413/b89d8626-d9a8-43fa-bdfc-acf92bda0422)

**Reply by acogoluegnes** [reply]
Yes, the channel is closed after the exchange declare error.

**Reply by michaelklishin** [reply]
[As the docs suggest](https://rabbitmq.com/channels.html) in the section on channel exceptions.

**Reply by Y-Meta** [reply]
get it Thanks！

**Comment by michaelklishin** [comment]
@Y-Meta unless you work in a software museum, the answer is [an upgrade is close to 10 years overdue](https://rabbitmq.com/versions.html).

**Reply by Y-Meta** [reply]
HAHAHA！yeah It's an ancient museum project；ok  I will upgrade it.  tks！
